tests/query/ConstantClausesUTest.cxxtest

```diff
@@ -403,8 +403,11 @@ void ConstantClausesUTest::test_complex()
 }
 
 /**
- * Constant clauses in a child atomspace are expected not to be accessible
- * from the parent atomspace (so the rewrite term should not be triggered).
+ * Constant clauses in a child atomspace would be expected not to be
+ * accessible from the parent atomspace, so the rewrite term would not
+ * be triggered, however constant clause are removed, regardless of
+ * whether they belong to the queried atomspace or not, thus, since
+ * empty is synonymous to true, the rewrite term is triggered.
```

Comment:
Actually, this is not correct. If nothing was found, then nothing was found, so nothing should be re-written and returned.

So that's a bug .... will note this in #2651